internal/apis/config/controller/validation/validation.go

```diff
@@ -23,71 +23,96 @@ import (
 	"net/url"
 	"strings"
 
-	//utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
 
 	config "github.com/cert-manager/cert-manager/internal/apis/config/controller"
 	defaults "github.com/cert-manager/cert-manager/internal/apis/config/controller/v1alpha1"
 )
 
-func ValidateControllerConfiguration(o *config.ControllerConfiguration) error {
-	if len(o.IngressShimConfig.DefaultIssuerKind) == 0 {
-		return errors.New("the --default-issuer-kind flag must not be empty")
+func ValidateControllerConfiguration(cfg *config.ControllerConfiguration) error {
+	var allErrors []error
+
+	if cfg.MetricsTLSConfig.FilesystemConfigProvided() && cfg.MetricsTLSConfig.DynamicConfigProvided() {
+		allErrors = append(allErrors, fmt.Errorf("invalid configuration: cannot specify both filesystem based and dynamic TLS configuration"))
+	} else {
+		if cfg.MetricsTLSConfig.FilesystemConfigProvided() {
+			if cfg.MetricsTLSConfig.Filesystem.KeyFile == "" {
+				allErrors = append(allErrors, fmt.Errorf("invalid configuration: tlsConfig.filesystem.keyFile (--metrics-tls-private-key-file) must be specified when using filesystem based TLS config"))
```

Comment:
I like the way you've included both the config file path and the CLI flag in the message. 
The existing validation functions seem quite inconsistent about that.